src/instance.rs

```diff
@@ -1329,11 +1329,8 @@ impl<T> Py<T> {
     /// # }
     /// ```
     #[inline]
-    pub fn clone_ref(&self, _py: Python<'_>) -> Py<T> {
-        unsafe {
-            ffi::Py_INCREF(self.as_ptr());
-            Self::from_non_null(self.0)
-        }
+    pub fn clone_ref(&self, py: Python<'_>) -> Py<T> {
+        unsafe { Py::from_borrowed_ptr(py, self.0.as_ptr()) }
```

Comment:
This doesn't seem related to deprecations to me. Should it land separately?

Reply:
Ah that's an unintended revert of #4313, will fix. Good catch